pyne/alara.py

```diff
@@ -943,3 +1116,214 @@ def _find_phsrc_dc(idc, phtn_src_dc):
         # if idc doesn't match any string in phtn_src_dc list, raise an error.
         raise ValueError(
             'Decay time {0} not found in phtn_src file'.format(idc))
+
+
+def responses_output_zone(responses=None, wdr_file=None, alara_params=None):
+    """
+    This function returns a string representing the output zone of alara input
+    code block.
+
+    Parameters
+    ----------
+    responses : list of string
+        A keyword represent the alara output zone. The following responses are
+        supported:
+            - decay_heat
+            - specific_activity
+            - alpha_heat
+            - beta_heat
+            - gamma_heat
+            - wdr
+            - photon_source
+    wdr_file : string
+        File name of the standard used to calculate wdr.
+    alara_params: string
+        Alara parameters
+
+    Returns
+    -------
+    String represent the output zone block corresponding to the response.
+    """
+
+    # set default value for functions do not need response
+    if responses == None:
+        return ''
+    # input check
+    for response in responses:
+        if response not in ('decay_heat', 'photon_source', 'specific_activity',
+                        'alpha_heat', 'beta_heat', 'gamma_heat', 'wdr'):
+            raise ValueError('response {0} not supported.'.format(response))
+
```

Comment:
I think a more streamlined version of this would be:
* create a dictionary of response -> string for output block
* start a list of strings with "output zone"
* for each response, add its string from the dictionary
* add "end" to the list of strings
* `'\n'.join()` the list

Reply:
you'll need to treat WDR and photon source specially

Reply:
Thanks. Suggestion adopted in https://github.com/pyne/pyne/pull/1123/commits/6997cf65c8bf97eb09781ff4645e750168820daa.